internal/pkg/aws/s3/s3.go

```diff
@@ -206,6 +209,78 @@ func (s *S3) isBucketExists(bucket string) (bool, error) {
 	return true, nil
 }
 
+// GetBucketTree retrieves the objects in an S3 bucket and creates an ASCII tree representing their folder structure.
+func (s *S3) GetBucketTree(bucket string) (string, error) {
+	exists, err := s.isBucket(bucket)
+	if err != nil {
+		return "", fmt.Errorf("unable to determine the existence of bucket %s: %w", bucket, err)
+	}
+	if !exists {
+		return "", nil
+	}
+
+	listResp := &s3.ListObjectsV2Output{}
+	for {
+		listParams := &s3.ListObjectsV2Input{
+			Bucket:            aws.String(bucket),
+			Delimiter:         aws.String(slashDelimiter),
+			ContinuationToken: listResp.NextContinuationToken,
+		}
+		listResp, err = s.s3Client.ListObjectsV2(listParams)
+		if err != nil {
+			return "", fmt.Errorf("list objects for bucket %s: %w", bucket, err)
+		}
+		if listResp.NextContinuationToken == nil {
+			break
+		}
+	}
+
+	tree := treeprint.New()
+	// Add top-level files.
+	for _, object := range listResp.Contents {
+		tree.AddNode(aws.StringValue(object.Key))
+	}
+	// Recursively add folders and their children.
+	if err := s.addNodes(tree, listResp.CommonPrefixes, bucket); err != nil {
+		return "", err
+	}
+	return tree.String(), nil
+}
+
+func (s *S3) addNodes(tree treeprint.Tree, prefixes []*s3.CommonPrefix, bucket string) error {
```

Comment:
nit: what do you think of passing a pointer? Probably could make the function calls slightly less heavy
```suggestion
func (s *S3) addNodes(tree *treeprint.Tree, prefixes []*s3.CommonPrefix, bucket string) error {
```

Reply:
discussed offline